lib/ioc-rc

```diff
@@ -543,10 +543,12 @@ __rc_jails () {
         local name="$(zfs get -H -o value org.freebsd.iocage:host_hostuuid \
                     $jail)"
         local boot="$(zfs get -H -o value org.freebsd.iocage:boot $jail)"
+        local _jail_hostid="$(zfs get -H -o value org.freebsd.iocage:hostid \
+                        $jail)"
         local priority="$(zfs get -H -o value org.freebsd.iocage:priority \
                         $jail)"
 
-        if [ "$boot" == "on" ] ; then
+        if [ "$boot" == "on" ] && [ "$_jail_hostid" == "$hostid" ] ; then
```

Comment:
Is there a reason you are going with the `test` convention vs using `&&` and `||`?
From what I understand, I believe `-a` and `-o` were "deprecated" from the posix standpoint (at least at one point).

from: http://pubs.opengroup.org/onlinepubs/9699919799/utilities/test.html

> The XSI extensions specifying the -a and -o binary primaries and the '(' and ')' operators have been
> marked obsolescent. (Many expressions using them are ambiguously defined by the grammar 
> depending on the specific expressions being evaluated.) Scripts using these expressions should be
> converted to the forms given below. Even though many implementations will continue to support these
> obsolescent forms, scripts should be extremely careful when dealing with user-supplied input that could
> be confused with these and other primaries and operators. Unless the application developer knows all
> the cases that produce input to the script, invocations like:
>
>    test "$1" -a "$2"
>
> should be written as:
> 
>     test "$1" && test "$2"

Not saying that your choice is wrong, or disagreeing with it. Just wondering if you could discuss why you chose to go that direction.